### ISSUE TITLE
False positives for Vertical Whitespace Violation all over the project - sample project attached

### ISSUE THREAD

**Issue by mortenholmgaard** [realm/SwiftLint#3257]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

All over my project it get this lint error: 
> Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 3. (vertical_whitespace)

But I am yet to find a location where it is true. Always "Currently 3" but this is shown at all kinds of random locations where there is zero or one line of vertical spacing. 
See an example here:
![Skærmbillede 2020-07-02 kl  21 49 38](https://user-images.githubusercontent.com/1279671/86403425-fb516100-bcad-11ea-84f7-63bc02294a85.png)
I have tried removing surrounding characters and reformatting but the error keeps coming, and I have also looked at the file without finding any special characters. I just can't figure out why that is happening in this project as we have other projects where it is just working fine. I have reformatted all of the project to 4 spaces and not tabs but with no change.

I have made a Test project to see if the problem persists in a new project and it does. The code may not make that much sense, but the case of the errors are clearly shown: [Test.zip](https://github.com/realm/SwiftLint/files/4866339/Test.zip)

And a similar problem with `trailing_newline` rule if enabled. In files like UIApplicationExtensions from the Test project, every thing seems fine with the trailing newline in the file, but we still get that error.

##### Complete output when running SwiftLint, including the stack trace and command used

We are running SwiftLint in a build phase:

> ${PODS_ROOT}/SwiftLint/swiftlint --strict

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
 **0.39.2**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? 
**CocoaPods**
* Paste your configuration file:

```yml
line_length:
  warning: 150
  ignores_comments: true
  ignores_interpolated_strings: true
  ignores_urls: true
excluded:
  - Pods
disabled_rules:
  - trailing_whitespace
  - todo
  - valid_ibinspectable
  - force_cast
  - force_try
  - trailing_newline #temp!!
  - large_tuple
  - no_fallthrough_only
  - for_where
  - file_length
identifier_name:
  excluded:
    - i
    - id
    - to
    - !str on
    - x
    - y
    - r
    - g
    - b
    - a
    - fn
  max_length: 50
type_name:
  excluded:
    - T
function_parameter_count: 8
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
**No**
* Which Xcode version are you using (check `xcodebuild -version`)?
**Version 11.5 (11E608c)**
* Do you have a sample that shows the issue?
**See sample project:** [Test.zip](https://github.com/realm/SwiftLint/files/4866339/Test.zip)

**Comment by mortenholmgaard** [comment]
Just found out that the reason properly is LF vs CRLF - and SwiftLint only seems to properly support LF. 
I just found an old issue about it but it has sadly not been fixed yet: https://github.com/realm/SwiftLint/issues/1786
